### PR TITLE
go: update to build on macOS

### PR DESCRIPTION
**Pull request by peterjgilbert**
This PR factors out some OS-specific syscall parameters for conditional compilation.

_This does not change the fact that **macOS is not officially supported by oasis-core**._

### CODE REVIEW

File: go/common/crypto/signature/signers/plugin/plugin.go [hunk 1]
```diff
@@ -77,26 +77,9 @@ func NewFactory(config interface{}, roles ...signature.SignerRole) (signature.Si
...
-	// do one or more of:
-	//
-	//  1. Get rid of the ~180 instances of `os.Exit`
-	//  2. Complain to the Go developers to make `plugin` not suck.
-	//  3. Complain to their OS vendor to implement PR_SET_PDEATH_SIG`.
-	//  4. Complain to Docker to make Docker not suck (reaping zombies
-	//     isn't that hard).
-	//
-	// We do use managed plugins so that in the graceful exit case,
-	// the cleanup will be done at least.
```
Comment (kostko) [inline go/common/crypto/signature/signers/plugin/plugin.go:95]: Why did you remove this, the comment still applies, for Mac OS specifically:
> //  3. Complain to their OS vendor to implement PR_SET_PDEATH_SIG`.

:grin:

File: go/common/syscall/syscall_darwin.go [hunk 2]
```diff
@@ -0,0 +1,9 @@
+// Package syscall defines OS-specific syscall parameters.
+package syscall
+
+import "syscall"
+
+const IoctlTermiosGetAttr = syscall.TIOCGETA
+
+// CmdAttrs is the SysProcAttr that will ensure graceful cleanup (on Linux).
```
Comment (kostko) [inline go/common/syscall/syscall_darwin.go:8]: Maybe add a comment saying that `PR_SET_PDEATH_SIG` is not implemented on Darwin and so this is empty and as a consequence the plugin subprocesses may not always be cleaned up.

File: go/common/syscall/syscall_darwin.go [hunk 3]
```diff
@@ -0,0 +1,9 @@
+// Package syscall defines OS-specific syscall parameters.
+package syscall
+
+import "syscall"
+
+const IoctlTermiosGetAttr = syscall.TIOCGETA
```
Comment (kostko) [inline go/common/syscall/syscall_darwin.go:6]: Add documentation comment.

File: go/common/syscall/syscall_linux.go [hunk 4]
```diff
@@ -0,0 +1,13 @@
+// Package syscall defines OS-specific syscall parameters.
+package syscall
+
+import "syscall"
+
+const IoctlTermiosGetAttr = syscall.TCGETS
```
Comment (kostko) [inline go/common/syscall/syscall_linux.go:6]: Add documentation comment.

File: go/common/crypto/signature/signers/plugin/plugin.go (hunk 1 above)
Comment (Yawning) [reply go/common/crypto/signature/signers/plugin/plugin.go:95]: Yeah, all of this comment is still applicable.

Comment (peterjgilbert) [reply go/common/crypto/signature/signers/plugin/plugin.go]: Restored 😄

**Review by kostko — APPROVED** [review]
